### PR TITLE
IGNITE-16121 Thin client: Implement KeyValueView

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Extract record serialization logic into `ClientRecordSerializer`.
* Implement `KeyValueView` in thin client.

### CODE REVIEW

**Review by isapego — APPROVED** [review]
Looks good to me.

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+    /** {@inheritDoc} */
+    @Override
+    public V get(@Nullable Transaction tx, @NotNull K key) {
+        return getAsync(tx, key).join();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public @NotNull CompletableFuture<V> getAsync(@Nullable Transaction tx, @NotNull K key) {
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java:83]: Done.

File: modules/client/src/test/java/org/apache/ignite/client/ClientKeyValueBinaryViewTest.java [hunk 2]
```diff
@@ -89,6 +89,19 @@ public void testPutGet() {
...
     }
 
+    @Test
+    public void testPutNull() {
+        Table table = defaultTable();
+        KeyValueView<Tuple, Tuple> kvView = table.keyValueView();
+
+        Tuple key = defaultTupleKey();
+
+        kvView.put(null, key, null);
```
Comment (AMashenkov) [inline modules/client/src/test/java/org/apache/ignite/client/ClientKeyValueBinaryViewTest.java:99]: null-tuple is not allowed.

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordSerializer.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        }
+    }
+
+    public Collection<R> readRecsNullable(ClientSchema schema, ClientMessageUnpacker in) {
+        return readRecs(schema, in, true, TuplePart.KEY_AND_VAL);
+    }
+
+    public Collection<R> readRecs(ClientSchema schema, ClientMessageUnpacker in) {
+        return readRecs(schema, in, false, TuplePart.KEY_AND_VAL);
+    }
```
Comment (AMashenkov) [inline modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordSerializer.java:126]: Let's drop if these methods are not used.

File: modules/client/src/test/java/org/apache/ignite/client/ClientKeyValueBinaryViewTest.java (hunk 2 above)
Comment (ptupitsyn) [reply modules/client/src/test/java/org/apache/ignite/client/ClientKeyValueBinaryViewTest.java:99]: It is allowed, please see https://github.com/apache/ignite-3/blob/825a20586414c5d5bbf38686ccd3de54c9189b48/modules/table/src/test/java/org/apache/ignite/internal/table/KeyValueBinaryViewOperationsTest.java#L84

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordSerializer.java (hunk 3 above)
Comment (ptupitsyn) [reply modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordSerializer.java:126]: Fixed.

**Review by AMashenkov — APPROVED** [review]

**Pull request merged**